fastly/kinesis.go

```diff
@@ -0,0 +1,238 @@
+package fastly
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"time"
+)
+
+// Kinesis represents a Kinesis response from the Fastly API.
+type Kinesis struct {
+	ServiceID string `mapstructure:"service_id"`
+	Version   int    `mapstructure:"version"`
+
+	Name              string     `mapstructure:"name"`
+	StreamName        string     `mapstructure:"topic"`
+	Region            string     `mapstructure:"region"`
+	AccessKey         string     `mapstructure:"access_key"`
+	SecretKey         string     `mapstructure:"secret_key"`
+	Format            string     `mapstructure:"format"`
+	FormatVersion     uint       `mapstructure:"format_version"`
+	ResponseCondition string     `mapstructure:"response_condition"`
+	Placement         string     `mapstructure:"placement"`
+	CreatedAt         *time.Time `mapstructure:"created_at"`
+	UpdatedAt         *time.Time `mapstructure:"updated_at"`
+	DeletedAt         *time.Time `mapstructure:"deleted_at"`
+}
+
+// kinesesByName is a sortable list of Kineses.
+type kinesesByName []*Kinesis
+
+// Len, Swap, and Less implement the sortable interface.
+func (s kinesesByName) Len() int      { return len(s) }
+func (s kinesesByName) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s kinesesByName) Less(i, j int) bool {
+	return s[i].Name < s[j].Name
+}
+
+// ListKinesesInput is used as input to the ListKineses function.
+type ListKinesesInput struct {
+	// Service is the ID of the service (required).
+	Service string
+
+	// Version is the specific configuration version (required).
+	Version int
+}
+
+// ListKineses returns the list of Kineses for the configuration version.
+func (c *Client) ListKineses(i *ListKinesesInput) ([]*Kinesis, error) {
+	if i.Service == "" {
+		return nil, ErrMissingServiceID
+	}
+
+	if i.Version == 0 {
+		return nil, ErrMissingServiceVersion
+	}
+
+	path := fmt.Sprintf("/service/%s/version/%d/logging/kinesis", i.Service, i.Version)
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var kineses []*Kinesis
+	if err := decodeBodyMap(resp.Body, &kineses); err != nil {
+		return nil, err
+	}
+	sort.Stable(kinesesByName(kineses))
```

Comment:
In general the stable sort is a sort where the ordering of elements determined to be equal is maintained in the output, but for this particular case I can't say why a stable sort was chosen. It may have been the case at some point in the past that the logging endpoint names were not always unique and that is why `Stable` was used here, but I'm not really sure.